sdk/documenttranslation/azure-ai-documenttranslation/samples/async_samples/sample_batch_translation_async.py

```diff
@@ -0,0 +1,101 @@
+# ------------------------------------
+
+import os
+import asyncio
+
+
+class BatchTranslationSampleAsync(object):
+
+    async def batch_translation_async():
+        # import libraries
+        from azure.core.credentials import AzureKeyCredential
+        from azure.ai.textanalytics.aio import DocumentTranslationClient
+        from azure.ai.documenttranslation import (
+            BatchDocumentInput,
+            StorageTarget
+        )
+
+        # get service secrets
+        endpoint = os.environ["AZURE_DOCUMENT_TRANSLATION_ENDPOINT"]
+        key = os.environ["AZURE_DOCUMENT_TRANSLATION_KEY"]
+        source_container_url = os.environ["AZURE_SOURCE_CONTAINER_URL"]
+        target_container_url_es = os.environ["AZURE_TARGET_CONTAINER_URL_ES"]
+        target_container_url_fr = os.environ["AZURE_TARGET_CONTAINER_URL_FR"]
+
+        # create service client
+        client = DocumentTranslationClient(endpoint, AzureKeyCredential(key))
+
+        # prepare translation job input
+        batch = [
+            BatchDocumentInput(
+                source_url=source_container_url,
+                source_language="en",
+                targets=[
+                    StorageTarget(
+                        target_url=target_container_url_es,
+                        language="es"
+                    ),
+                    StorageTarget(
+                        target_url=target_container_url_fr,
+                        language="fr"
+                    )
+                ],
+                storage_type="file"
+            )
+        ]
+
+        # run translation job
+        async with client:
+            job_detail = await client.create_translation_job(batch)
+
+            print("Job initial status: {}".format(job_detail.status))
+            print("Number of translations on documents: {}".format(job_detail.documents_total_count))
+
+            # get job result
+            job_result = await client.wait_until_done(job_detail.id)  # type: JobStatusDetail
+            if job_result.status == "Succeeded":
+                print("We translated our documents!")
+                if job_result.documents_failed_count > 0:
+                    check_documents(client, job_result.id)
+
+            if job_result.status in ["Failed", "ValidationFailed"]:
+                if job_result.error:
+                    print("Translation job failed: {}: {}".format(job_result.error.code, job_result.error.message))
+                check_documents(client, job_result.id)
+                exit(1)
+
+
+    def check_documents(client, job_id):
+        from azure.core.exceptions import ResourceNotFoundError
+
+        try:
+            doc_statuses = client.list_documents_statuses(job_id)  # type: ItemPaged[DocumentStatusDetail]
+        except ResourceNotFoundError as err:
+            print("Failed to process any documents in source/target container.")
+            raise err
+
+        docs_to_retry = []
+        for document in doc_statuses:
```

Comment:
```suggestion
        async for document in doc_statuses:
```

Reply:
this method is synchronous in the async client
`    @distributed_trace
    def list_documents_statuses(self, job_id, **kwargs):
        # type: (str, **Any) -> AsyncItemPaged[DocumentStatusDetail]
        """

        :param job_id: guid id for job
        :type job_id: str
        :keyword int results_per_page:
        :keyword int skip:
        :rtype: ~azure.core.paging.AsyncItemPaged[DocumentStatusDetail]
        """

        return self._client.document_translation.get_operation_documents_status(job_id, **kwargs)`

Reply:
Yes, but it returns the type [AsyncItemPaged](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/core/azure-core/azure/core/async_paging.py#L118) which will return an async iterable of items. The `async for` lets us loop through this.